src/rules/require-store-callbacks-use-set-param.ts

```diff
@@ -0,0 +1,40 @@
+import { createRule } from "../utils"
+import { extractStoreReferences } from "./reference-helpers/svelte-store"
+
+export default createRule("require-store-callbacks-use-set-param", {
+  meta: {
+    docs: {
+      description: "store callbacks must use `set` param",
+      category: "Possible Errors",
+      recommended: false,
+    },
+    schema: [],
+    messages: {
+      unexpected: "Store callbacks must use `set` param.",
+    },
+    type: "suggestion",
+  },
+  create(context) {
+    return {
+      Program() {
+        for (const { node } of extractStoreReferences(context, [
+          "readable",
+          "writable",
+        ])) {
+          const [_, fn] = node.arguments
+          if (!fn || fn.type !== "ArrowFunctionExpression") {
```

Comment:
I think it would be even better if we also check FunctionExpression.

```suggestion
          if (!fn || (fn.type !== "ArrowFunctionExpression" && fn.type !== "FunctionExpression")) {
```